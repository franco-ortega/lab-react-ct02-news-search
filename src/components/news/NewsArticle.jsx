import React from 'react';
import PropTypes from 'prop-types';

const NewsArticle = ({ author, title, description }) => (
  <>
    <h3>{title}</h3>
    <p>by {author}</p>
    <p>{description}</p>
  </>
);

NewsArticle.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string,
  description: PropTypes.string.isRequired
};

export default NewsArticle;
