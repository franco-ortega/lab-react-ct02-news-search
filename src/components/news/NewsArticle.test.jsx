import React from 'react';
import { render, cleanup } from '@testing-library/react';
import NewsArticle from './NewsArticle';
// global.fetch = require('node-fetch');

describe('NewsArticle component', () => {
  afterEach(() => cleanup());
  it('renders NewsArticle', () => {
    const { asFragment } = render(<NewsArticle
      title="a title"
      article="this article"
      description="that description" />);
    expect(asFragment()).toMatchSnapshot();
  });

});

