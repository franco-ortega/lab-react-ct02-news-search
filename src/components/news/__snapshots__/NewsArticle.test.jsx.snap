// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle component renders NewsArticle 1`] = `
<DocumentFragment>
  <h3>
    a title
  </h3>
  <p>
    by 
  </p>
  <p>
    that description
  </p>
</DocumentFragment>
`;
